client/js/components/shrink-stories-nav.js

```diff
@@ -1,42 +1,53 @@
 import { featureTest } from '../util';
 import { setStickyNavHeight } from '../reducers/sticky-nav-height';
 import { onWindowScrollThrottle$, onWindowScrollDebounce$ } from '../utils/dom-events';
+import fastdom from '../utils/fastdom-promise';
 
 const shrinkStoriesNav = (el, dispatch) => {
   if (!featureTest('position', 'sticky')) return;
 
-  const getIsNarrow = () => {
-    return el.classList.contains('numbered-list--horizontal-narrow');
-  };
-  const elFromTop = el.offsetTop;
-  const elHeight = el.offsetHeight;
-  const distanceScrolled = () => {
-    return window.pageYOffset;
-  };
-  const isScrolledEnough = () => {
-    return (distanceScrolled() > elFromTop + elHeight);
-  };
-  const setIsNarrow = (value) => {
-    if (value && isScrolledEnough() && !getIsNarrow()) {
-      el.classList.add('numbered-list--horizontal-narrow');
-
-      dispatch(setStickyNavHeight(el.offsetHeight));
-    } else if (!value && getIsNarrow()) {
-      el.classList.remove('numbered-list--horizontal-narrow');
-
-      dispatch(setStickyNavHeight(el.offsetHeight));
-    }
-  };
-
-  onWindowScrollThrottle$.subscribe({
-    next: () => setIsNarrow(distanceScrolled() > elFromTop)
+  const elFromTop = fastdom.measure(() => {
+    return el.offsetTop;
   });
-
-  onWindowScrollDebounce$.subscribe({
-    next: () => setIsNarrow(distanceScrolled() > elFromTop)
+  const elHeight = fastdom.measure(() => {
+    return el.offsetHeight;
   });
 
-  dispatch(setStickyNavHeight(el.offsetHeight));
+  Promise.all([elFromTop, elHeight]).then(values => {
+    const fromTop = values[0];
+    const height = values[1];
+
```

Comment:
There's quite a lot of DOM reading and writing in here too? Should we not be wrapping that as well?